docs/overview.md

```diff
@@ -0,0 +1,140 @@
+# WooCommerce Subscriptions Core Overview
+
+This document provides an overview of the WooCommerce Subscriptions Core codebase, designed to help developers understand the architecture, key components, and flows.
+
+## Introduction
+
+WooCommerce Subscriptions Core is a library that provides subscription functionality for WooCommerce. It allows store owners to sell products with recurring payments, handling the full subscription lifecycle including creation, renewals, cancellations, and more.
+
+## Architecture
+
+The codebase follows a modular, object-oriented architecture that integrates with WooCommerce. Here's the high-level structure:
+
+1. **Core Subscription Object**: `WC_Subscription` extends `WC_Order` and represents a subscription with all its properties and methods.
+
+2. **Main Plugin Class**: `WC_Subscriptions_Core_Plugin` handles initialization, hooks, and manages the plugin's lifecycle.
+
+3. **Functional Organization**: Most functionality is organized into classes by feature with supporting functions in dedicated files.
+
+## Key Components
+
+### 1. Subscription Data Model
+
+- `WC_Subscription` - Core class that represents a subscription
+- `WC_Subscriptions_Order` - Manages the relationship between orders and subscriptions
+- Data is stored in the `wp_posts` table as a custom `shop_subscription` post type with meta data in `wp_postmeta`
```

Comment:
Or is stored within the dedicated orders tables, which is the current default.

Reply:
Good catch! I replaced it with the link to HPOS.